tests/surrogate_test.py

```diff
@@ -46,12 +59,18 @@ def test_initialization_1d():
         ChebyshevFirstKind.make_nested_set(2))
     domain = (-1, 1)
     surrogate = Surrogate(domain, grid_generator)
+    gradient_surrogate = GradientSurrogate(domain, grid_generator)
```

Comment:
`Surrogate` and `GradientSurrogate` should be tested separately, write individual test functions for each.